pennylane/templates/subroutines/qpe.py

```diff
@@ -104,19 +108,68 @@ def circuit():
 
             # Need to rescale phase due to convention of RX gate
             phase_estimated = 4 * np.pi * (1 - phase_estimated)
+
+        We can also perform phase estimation on an operator. Note that since operators are defined
+        with target wires, the target wires should not be provided for the PQE.
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=2)
+
+            # use the product to specify compound operators
+            unitary = qml.RX(np.pi / 2, wires=[0]) @ qml.CNOT(wires=[0, 1])
+            eigenvector = np.array([-1/2, -1/2, 1/2, 1/2])
+
+            n_estimation_wires = 5
+            estimation_wires = range(2, n_estimation_wires + 2)
+            target_wires = [0, 1]
+
+            dev = qml.device("default.qubit", wires=n_estimation_wires + 2)
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.QubitStateVector(eigenvector, wires=target_wires)
+                QuantumPhaseEstimation(
+                    unitary,
+                    estimation_wires=estimation_wires,
+                )
+                return qml.probs(estimation_wires)
+
+            phase_estimated = np.argmax(circuit()) / 2 ** n_estimation_wires
+
     """
     num_wires = AnyWires
     grad_method = None
 
-    def __init__(self, unitary, target_wires, estimation_wires, do_queue=True, id=None):
+    def __init__(self, unitary, target_wires=None, estimation_wires=None, do_queue=True, id=None):
+
+        if isinstance(unitary, Operator):
+
+            # If the unitary is expressed in terms of operators, do not provide target wires
+            if target_wires is not None:
+                raise qml.QuantumFunctionError(
+                    "The unitary is expressed as an operator, which already has target wires "
+                    "defined, do not additionally specify target wires."
+                )
+            target_wires = unitary.wires
+
+        elif target_wires is None:
```

Comment:
To avoid having to check between the two cases (matrix versus operator) downstream, we could create a `QubitUnitary` operator here to unify the two cases.

Then `unitary` would always be a `Operator` after initialization.  We could then store the operator in the hyperparameters.

This would also mean we would need a data getter and setter like in [`SymbolicOp`](https://github.com/PennyLaneAI/pennylane/blob/6881ca8076983f4f324492cfeb8edb10ec06a5be/pennylane/ops/op_math/symbolicop.py#L79). At least if we still want to treat the `QPE` data like the target operator data.

Reply:
Good idea! I will implement it now.

Reply:
If we wrap the unitary matrix in a `QubitUnitary`, then the controlled unitary would be implemented with `qml.ctrl` which returns a general `Controlled` operator instead of `ControlledQubitUnitary`. This is completely fine I guess, but the `ControlledQubitUnitary` class would seem a bit redundant after this change.

Reply:
Given `ControlledQubitUnitary` probably has less features than `ControlledOp`, I'm not too concerned.  We'll probably be replacing most of it soon anyway.